include/rocksdb/advanced_options.h

```diff
@@ -684,6 +684,17 @@ struct AdvancedColumnFamilyOptions {
   // Dynamically changeable through SetOptions() API
   uint64_t max_compaction_bytes = 0;
 
+  // When setting up compaction input files, we ignore the
+  // `max_compaction_bytes` limit when pulling in input files that are entirely
+  // within output key range.
+  //
+  // Default: true
+  //
+  // Dynamically changeable through SetOptions() API
+  // We could remove this knob and always ignore the limit once it is proven
+  // safe.
```

Comment:
Our team isn't well-known for our ability to remove things from the public API once we no longer need them, but I'm fine with trying it.

Reply:
Thanks for the review.

> Our team isn't well-known for our ability to remove things from the public API once we no longer need them

I'll set a reminder for this..